src/models/projectDetector/projectInfo.ts

```diff
@@ -0,0 +1,4 @@
+export interface IProjectInfo {
+  name?: string;
```

Comment:
Instead of using a plain string I think we could use a union type of string literal types or an enum.

Reply:
You mean like an enum of project names?

i.e.
```
enum Projects {
  none: ''
  angular: 'ng',
  angularjs: 'ngjs'
}
```

Reply:
yes, that or this:
```ts
export type Project = 'ng' | 'ngjs' | 'react';
// and then

interface IProjectInfo {
name?: Project;
``` 

I don't have any preference.

PS: In case anyone is not familiar with `string literal types` you can have more info [here](https://basarat.gitbooks.io/typescript/docs/types/literal-types.html)

Reply:
I'm familiar but I think `enum` gives more flexibility. Will try them and see what fits best.

Reply:
I think this should be a `Projects` enum